src/end_entity.rs

```diff
@@ -69,15 +68,6 @@ impl<'a> TryFrom<&'a CertificateDer<'a>> for EndEntityCert<'a> {
 }
 
 impl<'a> EndEntityCert<'a> {
-    /// Verifies that the end-entity certificate is valid for use against the
-    /// specified Extended Key Usage (EKU).
-    ///
-    /// `time` is the time for which the validation is effective (usually the
-    /// current time).
-    pub fn verify_for_usage(&self, time: UnixTime, opts: &ChainOptions<'_>) -> Result<(), Error> {
```

Comment:
I'm not sure about this one. Can you talk more about the motivation for this commit?

To me it feels more intuitive to be calling a method on an `EndEntityCert` providing options, instead of calling a method on the options providing a cert. In my experience it's more common for an API to associate this with the cert object. [Go does](https://pkg.go.dev/crypto/x509#Certificate.Verify) as one quick example, and I think most C APIs take a cert-like object as the first arg and then trail the options.

Reply:
I don't feel strongly about this particular change, so we could just chose to remove it.

To me this setup feels more idiomatic, perhaps because of how the lifetimes work out? Typically the trust anchors and the intermediates will have a long lifetime and the `EndEntityCert` has a shorter lifetime. In a way, it feels more like a "context" and less like "options". One might also think about how a rustls verifier implementation could hold on to an owned variant of this.

Reply:
> Typically the trust anchors and the intermediates will have a long lifetime and the EndEntityCert has a shorter lifetime.

That makes sense and seems like a good reason to accept the unfamiliarity of the API. For trust anchors this feels obvious. For intermediates I had to think about it more, but its not unheard of to cache intermediates in a way that ends up living longer than what might have been read from the handshake alongside the EE cert.

> In a way, it feels more like a "context" and less like "options".

This feels like a hint towards a better name for the struct since you don't love the current one. Maybe `ChainContext` would be nicer?

Reply:
I've cut this part out of the commits.